File: src/dbflute/adf/meta/vendorPartMandbm.go
package meta

import (
	"github.com/mikeshimura/dbflute/df"
)

type VendorPartManDbm_T struct {
	df.BaseDBMeta
	ColumnPartManId *df.ColumnInfo
	ColumnPartManName *df.ColumnInfo
	ColumnPartManPoint *df.ColumnInfo
	ColumnPartManDate *df.ColumnInfo
}

func (b *VendorPartManDbm_T) GetProjectName() string {
	return df.DBCurrent_I.ProjectName
}

func (b *VendorPartManDbm_T) GetDbCurrent() *df.DBCurrent {
	return df.DBCurrent_I
}

var VendorPartManDbm *VendorPartManDbm_T

func Create_VendorPartManDbm() {
	VendorPartManDbm = new(VendorPartManDbm_T)
	VendorPartManDbm.TableDbName = "vendor_part_man"
	VendorPartManDbm.TableDispName = "vendor_part_man"
	VendorPartManDbm.TablePropertyName = "vendorPartMan"
	VendorPartManDbm.TableSqlName = new(df.TableSqlName)
	VendorPartManDbm.TableSqlName.TableSqlName = "vendor_part_man"
	VendorPartManDbm.TableSqlName.CorrespondingDbName = VendorPartManDbm.TableDbName

	var vendorPartMan df.DBMeta
	vendorPartMan = VendorPartManDbm
	VendorPartManDbm.DBMeta=&vendorPartMan
	partManIdSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo part_man_id
	partManIdSqlName.ColumnSqlName = "part_man_id"
	partManIdSqlName.IrregularChar = false
	VendorPartManDbm.ColumnPartManId = df.CCI(&vendorPartMan, "part_man_id", partManIdSqlName, "", "", "Integer.class", "partManId", "", true, false,true, "int4", 10, 0, "",false,"","", "","","",false,"int64")
	partManNameSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo part_man_name
	partManNameSqlName.ColumnSqlName = "part_man_name"
	partManNameSqlName.IrregularChar = false
	VendorPartManDbm.ColumnPartManName = df.CCI(&vendorPartMan, "part_man_name", partManNameSqlName, "", "", "String.class", "partManName", "", false, false,true, "varchar", 2147483647, 0, "",false,"","", "","","",false,"string")
	partManPointSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo part_man_point
	partManPointSqlName.ColumnSqlName = "part_man_point"
	partManPointSqlName.IrregularChar = false
	VendorPartManDbm.ColumnPartManPoint = df.CCI(&vendorPartMan, "part_man_point", partManPointSqlName, "", "", "Integer.class", "partManPoint", "", false, false,true, "int4", 10, 0, "",false,"","", "","","",false,"int64")
	partManDateSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo part_man_date
	partManDateSqlName.ColumnSqlName = "part_man_date"
	partManDateSqlName.IrregularChar = false
	VendorPartManDbm.ColumnPartManDate = df.CCI(&vendorPartMan, "part_man_date", partManDateSqlName, "", "", "java.time.LocalDate.class", "partManDate", "", false, false,false, "date", 13, 0, "",false,"","", "","","",false,"df.NullDate")

	VendorPartManDbm.ColumnInfoList = new(df.List)
	VendorPartManDbm.ColumnInfoList.Add(VendorPartManDbm.ColumnPartManId)
	VendorPartManDbm.ColumnInfoList.Add(VendorPartManDbm.ColumnPartManName)
	VendorPartManDbm.ColumnInfoList.Add(VendorPartManDbm.ColumnPartManPoint)
	VendorPartManDbm.ColumnInfoList.Add(VendorPartManDbm.ColumnPartManDate)


	VendorPartManDbm.ColumnInfoMap=make(map[string]int)
	VendorPartManDbm.ColumnInfoMap["partManId"]=0
		VendorPartManDbm.ColumnInfoMap["partManName"]=1
		VendorPartManDbm.ColumnInfoMap["partManPoint"]=2
		VendorPartManDbm.ColumnInfoMap["partManDate"]=3
	    VendorPartManDbm.PrimaryKey = true
    VendorPartManDbm.CompoundPrimaryKey = false
	ui := new(df.UniqueInfo)
	ui.DbMeta = &vendorPartMan
	ui.Primary = true
	ui.UniqueColumnList = new(df.List)
	ui.UniqueColumnList.Add(VendorPartManDbm.ColumnPartManId)

	VendorPartManDbm.PrimaryInfo = new(df.PrimaryInfo)
	VendorPartManDbm.PrimaryInfo.UniqueInfo = ui
	
	var vendorPartManMeta df.DBMeta = VendorPartManDbm
	df.DBMetaInstanceHandler_I.TableDbNameInstanceMap["VendorPartMan"] = &vendorPartManMeta
}

File: src/main/java/com/mssoftech/dbflute/bsentity/dbmeta/VendorPartManDbm.java
package com.mssoftech.dbflute.bsentity.dbmeta;

import java.util.List;
import java.util.Map;

import org.dbflute.Entity;
import org.dbflute.dbmeta.AbstractDBMeta;
import org.dbflute.dbmeta.info.*;
import org.dbflute.dbmeta.name.*;
import org.dbflute.dbmeta.property.PropertyGateway;
import org.dbflute.dbway.DBDef;
import com.mssoftech.dbflute.allcommon.*;
import com.mssoftech.dbflute.exentity.*;

/**
 * The DB meta of vendor_part_man. (Singleton)
 * @author DBFlute(AutoGenerator)
 */
public class VendorPartManDbm extends AbstractDBMeta {

    // ===================================================================================
    //                                                                           Singleton
    //                                                                           =========
    private static final VendorPartManDbm _instance = new VendorPartManDbm();
    private VendorPartManDbm() {}
    public static VendorPartManDbm getInstance() { return _instance; }

    // ===================================================================================
    //                                                                       Current DBDef
    //                                                                       =============
    public String getProjectName() { return DBCurrent.getInstance().projectName(); }
    public String getProjectPrefix() { return DBCurrent.getInstance().projectPrefix(); }
    public String getGenerationGapBasePrefix() { return DBCurrent.getInstance().generationGapBasePrefix(); }
    public DBDef getCurrentDBDef() { return DBCurrent.getInstance().currentDBDef(); }

    // ===================================================================================
    //                                                                    Property Gateway
    //                                                                    ================
    // -----------------------------------------------------
    //                                       Column Property
    //                                       ---------------
    protected final Map<String, PropertyGateway> _epgMap = newHashMap();
    { xsetupEpg(); }
    protected void xsetupEpg() {
        setupEpg(_epgMap, et -> ((VendorPartMan)et).getPartManId(), (et, vl) -> ((VendorPartMan)et).setPartManId(cti(vl)), "partManId");
        setupEpg(_epgMap, et -> ((VendorPartMan)et).getPartManName(), (et, vl) -> ((VendorPartMan)et).setPartManName((String)vl), "partManName");
        setupEpg(_epgMap, et -> ((VendorPartMan)et).getPartManPoint(), (et, vl) -> ((VendorPartMan)et).setPartManPoint(cti(vl)), "partManPoint");
        setupEpg(_epgMap, et -> ((VendorPartMan)et).getPartManDate(), (et, vl) -> ((VendorPartMan)et).setPartManDate(ctld(vl)), "partManDate");
    }
    public PropertyGateway findPropertyGateway(String prop)
    { return doFindEpg(_epgMap, prop); }

    // ===================================================================================
    //                                                                          Table Info
    //                                                                          ==========
    protected final String _tableDbName = "vendor_part_man";
    protected final String _tableDispName = "vendor_part_man";
    protected final String _tablePropertyName = "vendorPartMan";
    protected final TableSqlName _tableSqlName = new TableSqlName("vendor_part_man", _tableDbName);
    { _tableSqlName.xacceptFilter(DBFluteConfig.getInstance().getTableSqlNameFilter()); }
    public String getTableDbName() { return _tableDbName; }
    public String getTableDispName() { return _tableDispName; }
    public String getTablePropertyName() { return _tablePropertyName; }
    public TableSqlName getTableSqlName() { return _tableSqlName; }

    // ===================================================================================
    //                                                                         Column Info
    //                                                                         ===========
    protected final ColumnInfo _columnPartManId = cci("part_man_id", "part_man_id", null, null, Integer.class, "partManId", null, true, false, true, "int4", 10, 0, null, false, null, null, null, null, null, false);
//"int64"
    protected final ColumnInfo _columnPartManName = cci("part_man_name", "part_man_name", null, null, String.class, "partManName", null, false, false, true, "varchar", 2147483647, 0, null, false, null, null, null, null, null, false);
//"string"
    protected final ColumnInfo _columnPartManPoint = cci("part_man_point", "part_man_point", null, null, Integer.class, "partManPoint", null, false, false, true, "int4", 10, 0, null, false, null, null, null, null, null, false);
//"int64"
    protected final ColumnInfo _columnPartManDate = cci("part_man_date", "part_man_date", null, null, java.time.LocalDate.class, "partManDate", null, false, false, false, "date", 13, 0, null, false, null, null, null, null, null, false);
//"df.NullDate"

    /**
     * part_man_id: {PK, NotNull, int4(10)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnPartManId() { return _columnPartManId; }
    /**
     * part_man_name: {NotNull, varchar(2147483647)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnPartManName() { return _columnPartManName; }
    /**
     * part_man_point: {NotNull, int4(10)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnPartManPoint() { return _columnPartManPoint; }
    /**
     * part_man_date: {date(13)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnPartManDate() { return _columnPartManDate; }

    protected List<ColumnInfo> ccil() {
        List<ColumnInfo> ls = newArrayList();
        ls.add(columnPartManId());
        ls.add(columnPartManName());
        ls.add(columnPartManPoint());
        ls.add(columnPartManDate());
        return ls;
    }

    { initializeInformationResource(); }

    // ===================================================================================
    //                                                                         Unique Info
    //                                                                         ===========
    // -----------------------------------------------------
    //                                       Primary Element
    //                                       ---------------
    protected UniqueInfo cpui() { return hpcpui(columnPartManId()); }
    public boolean hasPrimaryKey() { return true; }
    public boolean hasCompoundPrimaryKey() { return false; }

    // ===================================================================================
    //                                                                       Relation Info
    //                                                                       =============
    // cannot cache because it uses related DB meta instance while booting
    // (instead, cached by super's collection)
    // -----------------------------------------------------
    //                                      Foreign Property
    //                                      ----------------

    // -----------------------------------------------------
    //                                     Referrer Property
    //                                     -----------------

    // ===================================================================================
    //                                                                        Various Info
    //                                                                        ============

    // ===================================================================================
    //                                                                           Type Name
    //                                                                           =========
    public String getEntityTypeName() { return "com.mssoftech.dbflute.exentity.VendorPartMan"; }
    public String getConditionBeanTypeName() { return "com.mssoftech.dbflute.cbean.VendorPartManCB"; }
    public String getBehaviorTypeName() { return "com.mssoftech.dbflute.exbhv.VendorPartManBhv"; }

    // ===================================================================================
    //                                                                         Object Type
    //                                                                         ===========
    public Class<VendorPartMan> getEntityType() { return VendorPartMan.class; }

    // ===================================================================================
    //                                                                     Object Instance
    //                                                                     ===============
    public VendorPartMan newEntity() { return new VendorPartMan(); }

    // ===================================================================================
    //                                                                   Map Communication
    //                                                                   =================
    public void acceptPrimaryKeyMap(Entity et, Map<String, ? extends Object> mp)
    { doAcceptPrimaryKeyMap((VendorPartMan)et, mp); }
    public void acceptAllColumnMap(Entity et, Map<String, ? extends Object> mp)
    { doAcceptAllColumnMap((VendorPartMan)et, mp); }
    public Map<String, Object> extractPrimaryKeyMap(Entity et) { return doExtractPrimaryKeyMap(et); }
    public Map<String, Object> extractAllColumnMap(Entity et) { return doExtractAllColumnMap(et); }
}
